ladybug_geometry/geometry3d/face.py

```diff
@@ -1151,6 +1151,108 @@ def sub_faces_by_ratio_sub_rectangle(self, ratio, sub_rect_height, sill_height,
+                precedence and the sub-rectangle height will be larger than this value.
+            sub_face_width: A number for the target width of the output rectangles.
+                If the ratio is too large for the width, the ratio will take 
+                precedence and the sub-rectangle width will be larger than this value.
+            dist_breakup: A number for the target breakup separation distance between
+                individual sub-rectangle centerlines.  If this number is larger than
+                the parent rectangle base, only one sub-rectangle will be produced.
+            sill_height: A number for the target height above the bottom edge of
+                the rectangle to start the sub-rectangles. If the ratio is too
+                large for the height, the ratio will take precedence and the
+                sub-rectangle height will be smaller than this value.
+            base_plane: A Plane object in which the rectangle exists.
+                The origin of this plane will be the lower left corner of the
+                rectangle and the X and Y axes will form the sides.
+                Default is the world XY plane.
+
+        Returns:
+            A list of Face3D objects for sub faces.
+        """
+        # if sub_face_height > rect_height, set it to just under rect_height
+        sub_face_height_final = (
+            rect_height - 0.01 if sub_face_height > rect_height - 0.01
+            else sub_face_height)
+        # if sill_height is close to 0, set it to just above 0
+        sill_height_final = (
+            0.01 * rect_height if sill_height < 0.01 * rect_height else sill_height)
+        # adjust sill_height_final if sum of it and sub_face_height_final > rect_height
+        if sub_face_height_final + sill_height_final > rect_height:
+            sill_height_final = rect_height - sub_face_height_final
+
+        max_width_break_up = rect_base / 2
+        num_divisions = (
+            round(rect_base / dist_breakup) if rect_base > dist_breakup / 2 else 1)
+        if (sub_face_width < max_width_break_up and
+            num_divisions * sub_face_width <= rect_base):
+            # divide up the rectangle into points on the bottom
+            div_dist = rect_base / 2 if num_divisions == 1 else dist_breakup
+            if (num_divisions * sub_face_width + (num_divisions - 1) * \
+                (dist_breakup - sub_face_width) > rect_base):
+                num_divisions = math.floor(rect_base / dist_breakup)
+
+            is_odd = lambda num: num % 2
+            start_pt_x = (
+                (num_divisions / 2) * div_dist if is_odd(num_divisions) == 0
+                else (math.floor(num_divisions / 2) + 0.5) * div_dist)
+
+            btm_div_pts = [Point3D(start_pt_x, 0, sill_height_final)]
+            remainder = rect_base - (div_dist * num_divisions)
+            total_dist = remainder / 2
+
+            while total_dist < rect_base:
+                total_dist += div_dist
```

Comment:
I just realized that we could get thrown into an infinite loop here if someone inputs a negative number or 0 for `dist_breakup`. Maybe we should add an `assert` statement at the top of the function just to be sure that the `dist_breakup` is always positive.

Reply:
It looks like you addressed this one.